clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```
@@ -71,8 +72,13 @@ public JmxReporter() {
 
     /**
      * Create a JMX reporter that prefixes all metrics with the given string.
+     *  @deprecated Since 2.6.0. Use {@link JmxReporter#JmxReporter()}
+     *  Initialize JmxReporter with {@link JmxReporter#contextChange(MetricsContext)}
+     *  Populate prefix by adding _namespace/prefix key value pair to {@link MetricsContext}
      */
+    @Deprecated
     public JmxReporter(String prefix) {
+        Objects.requireNonNull(prefix);
```

Should we throw an exception here, or just replace the null value with an empty string?

---

Change the logic here: if null, set it to space.